### ISSUE TITLE
Imported font doesn't work - with warning in console

### ISSUE THREAD

**Issue by JeFawk** [Scirra/Construct-bugs#7273]
## Problem description
I imported the following font:
[Chalkduster.ttf.zip](https://github.com/Scirra/Construct-bugs/files/12214363/Chalkduster.ttf.zip)

First I noticed problems with the font not appearing in the font selection of a text object. Then somehow, it appeared in the list, but when running the game, it gives console warning and the font doesn't work, it reverts to something like Times New Roman.

The preview looks like so:
<img width="1680" alt="image" src="https://github.com/Scirra/Construct-bugs/assets/7268697/da40b6fd-c4a7-4226-bcb6-1ce4f1f9e5cb">

Other fonts I've imported seem to work.

## Attach a .c3p

[New project.c3p.zip](https://github.com/Scirra/Construct-bugs/files/12214360/New.project.c3p.zip)

## Steps to reproduce

1. open a blank project
2. import font
3. run game
4. alternatively set it as a font for a text object

## Observed result

Notice the text object didn't work, and the console has the warning above.

## More details

**Affected browsers/platforms:** Chrome macOS

**First affected release:** broken for at least 2-3 months

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r351 (beta)
Browser: Chrome 115.0.5790.114
Browser engine: Chromium
Context: browser
Operating system: macOS 12.6.0
Device type: desktop
Device pixel ratio: 2
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 276 gb
Storage usage (approx): 133 mb (0%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc. (Apple)
Renderer: ANGLE (Apple, Apple M1, OpenGL 4.1)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 64
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
WEBGL_provoking_vertex
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.265 (video/mp4; codecs=hev1.1.2.L93.B0)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM AV1 (video/webm; codecs=av1)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
I don't think this is a problem with Construct - it looks like an unsupported or corrupt font file (the browser log says `OTS parsing error: cmap: Failed to parse table`).

I'd suggest only using fonts in WOFF format. That is the main format that browsers support, and I think using any other formats is a roll of the dice if it works or not, and so I don't think we will investigate or support any issues arising from using non-WOFF formats.

**Comment by AshleyScirra** [comment]
I added a section on the recommended font file format to the [Best practices manual entry](https://www.construct.net/en/make-games/manuals/construct-3/tips-and-guides/best-practices).
